## Providing profitable App profiles for Developers
The aim of this project is to provide developers with application profiles that attract most users

In our company, we develop appps that are free to install and directed towards english-speaking audience and our main source of revenue consists of in-app ads

In [1]:
from csv import reader

def open_dataset(dataset):
    opened_file = open(dataset, encoding='utf8')
    read_file = reader(opened_file)
    data_list = list(read_file)
    data_header = data_list[0]
    data = data_list[1:]
    
    return data, data_header

In [2]:
apple_data, apple_header = open_dataset('AppleStore.csv')
google_data, google_header = open_dataset('googleplaystore.csv')

In [3]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for each_row in dataset_slice:
        print(each_row)
        print('\n')
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [4]:
print(apple_header)
print('\n')
apple_explrd = explore_data(apple_data, 0, 5, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37'

From this output, we can see that the AppStore dataset has 7197 apps; the columns that will be useful for the purpose of our analysis are `'id', 'track_name', 'currency', 'price', 'rating_count_tot', 'user_rating', 'prime_genre'`. Not all column names are descriptive enough, click [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) for detailed description.

Now for the Google Play Store dataset.

In [5]:
print(google_header)
print('\n')
google_explrd = explore_data(google_data, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

We can see that the Google Play Store dataset has 7197 apps; the columns that will be useful for the purpose of our analysis are `'App', 'Category', 'Rating', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres'`. Click [here](https://www.kaggle.com/lava18/google-play-store-apps) for detailed description of the column names.


## Deleting wrong data
Checking and deleting the data with the wrong rating.

In [6]:
print(google_data[10472])
print(google_header)

del google_data[10472]

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


## Duplicate data

### Part 1 - Affirming data duplicates
The Google Play Store dataset has some duplicate data. An example of duplicated data will be printed for confirmation.

In [7]:
for each_app in google_data:
   if each_app[0] == 'Instagram':
        print(each_app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


Now, we can see that the Instagram app is duplicated four times in the Google Play dataset, lets see how many duplicates we have in total.

In [8]:
google_duplicate_apps = []
google_unique_apps = []

for each_app in google_data:
    if each_app[0] in google_unique_apps:
        google_duplicate_apps.append(each_app[0])
    else:
        google_unique_apps.append(each_app[0])
        
print('Number of duplicated rows is: '+ str(len(google_duplicate_apps)))

Number of duplicated rows is: 1181


As seen, we have 1181 duplicated apps. Let's see how many times each app was duplicated.

In [9]:
google_dupl_dict = {}

for each_app in google_duplicate_apps:
    if each_app in google_dupl_dict:
        google_dupl_dict[each_app] += 1
    else:
        google_dupl_dict[each_app] = 1
        
print(google_dupl_dict)

{'Quick PDF Scanner + OCR FREE': 2, 'Box': 2, 'Google My Business': 2, 'ZOOM Cloud Meetings': 1, 'join.me - Simple Meetings': 2, 'Zenefits': 1, 'Google Ads': 2, 'Slack': 2, 'FreshBooks Classic': 1, 'Insightly CRM': 1, 'QuickBooks Accounting: Invoicing & Expenses': 2, 'HipChat - Chat Built for Teams': 1, 'Xero Accounting Software': 1, 'MailChimp - Email, Marketing Automation': 1, 'Crew - Free Messaging and Scheduling': 1, 'Asana: organize team projects': 1, 'Google Analytics': 1, 'AdWords Express': 1, 'Accounting App - Zoho Books': 1, 'Invoice & Time Tracking - Zoho': 1, 'Invoice 2go — Professional Invoices and Estimates': 1, 'SignEasy | Sign and Fill PDF and other Documents': 1, 'Genius Scan - PDF Scanner': 1, 'Tiny Scanner - PDF Scanner App': 1, 'Fast Scanner : Free PDF Scan': 1, 'Mobile Doc Scanner (MDScan) Lite': 1, 'TurboScan: scan documents and receipts in PDF': 1, 'Tiny Scanner Pro: PDF Doc Scan': 1, 'Docs To Go™ Free Office Suite': 1, 'OfficeSuite : Free Office + PDF Editor': 1,

This shows the number of times each app was duplicated. We need to remove this duplicates based on the number of reviews, that is, the duplicate with the highest number of reviews will be selected and others will be removed. This criterion is so because the duplicate with the highest number of reviews signifies the most recent data for that particular app.

The next step is to select the app entry with the highest number of reviews.

### Part 2 - Removing data duplicates

We created a dictionary that contains each unique app with their maximum number of reviews

In [10]:
google_reviews_max = {}

for each_app in google_data:
    name = each_app[0]
    n_reviews = float(each_app[3])
    
    if name in google_reviews_max:
        google_reviews_max[name] = max(google_reviews_max[name], n_reviews)
    else:
        google_reviews_max[name] = n_reviews
        
print(len(google_reviews_max))

9659


This shows that the we 9659 unique apps in the google dataset.

The unique apps with highest number of reviews in the google play store dataset are extracted to the `android_clean` list  and the `already_added` list is used to avoid duplicating apps that have more than one entry with same number of reviews

In [11]:
google_cleaned_data = []
already_added = []

for each_app in google_data:
    name = each_app[0]
    n_reviews = float(each_app[3])
    
    if (n_reviews == google_reviews_max[name]) and (name not in already_added):
        google_cleaned_data.append(each_app)
        already_added.append(each_app[0])

Exploring the cleaned google dataset and confirming that the number of rows is 9,659

In [12]:
explore_data(google_cleaned_data, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


The output shows that we have 9,659 rows as expected.



## Removing Non-English Apps

### Part 1
We need to define a function that will be used to test if the name of an app is english or not since or targeted audience is english-speaking.

But first, let's test the function out with some examples:

In [13]:
def check_if_english(app_name):
    for each_charac in app_name:
        charac_num = ord(each_charac)
        if charac_num > 127:
            return False
        
    return True
print(check_if_english('Instagram'))
print(check_if_english('a奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_if_english('Docs To Go™ Free Office Suite'))
print(check_if_english('Instachat 😜'))

True
False
False
False


Apparently, the `check_if_english` function seems to differentiate between a non-english and an english name, but some english names contain characters that are not within the ASCII range. Using this function as it is will make us remove useful data therefore, we need to modify the function.


### Part 2

We modify the `check_if_english` function to remove apps whose names have more than 3 characters whose corresponding number is out of ASCII range. Though, this function is not perfect but it will reduce the impact of data loss.

In [14]:
def check_if_english(app_name):
    flag = 0
    for each_charac in app_name:
        charac_num = ord(each_charac)
        if charac_num > 127:
            flag += 1
            if flag > 3:
                return False
        
    return True

print(check_if_english('Instagram'))
print(check_if_english('a奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_if_english('Docs To Go™ Free Office Suite'))
print(check_if_english('Instachat 😜'))

True
False
True
True


Filtering both datasets

In [15]:
google_english = []
apple_english = []

for each_app in google_cleaned_data:
    if check_if_english(each_app[0]):
        google_english.append(each_app)

for each_app in apple_data:
    if check_if_english(each_app[2]):
        apple_english.append(each_app)
        
explore_data(google_english, 0, 3, True)
explore_data(apple_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13
['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '18858

We have approximately 9,614 english google apps and 6,183 english apple apps in our datasets.

## Isolating Free Apps
The final step in the data cleaning process for this project is to isolate the free apps. Since the free apps in the google dataset have their values equal to `'0'` in the price column and the free apps in the apple dataset have their values equal to `0` in the price column, we can use this information to isolate them.

In [16]:
google_final = []
apple_final = []

for each_app in google_english:
    if each_app[7] == '0':
        google_final.append(each_app)
        
for each_app in apple_english:
    price = float(each_app[5])
    if price == 0:
        apple_final.append(each_app)
        
explore_data(google_final, 0, 3, True)
print('\n')
explore_data(apple_final, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Sh

We are left with 8864 google apps and 3222 android whcih is sufficient for our analysis.

## Most Apps By Genre

### Part 1
As mentioned earlier, the aim of this project is to determine the kind of app that will attract more users because our revenue generation is highly dependent on number of users for our app.

The process of validation for our app idea so as to reduce risk and overhead are:
    1. Develop a minimal version of the app and then upload it to Google Play Store.
    2. If the app gets enough postive respone from users, then we develop it further.
    3. If after six months the app is deemed to be profitable, then we develop an ios version of the app for App Store.
    
Since we are targetting both markets (App Store and Google Play Store), we need to analyse the profile of the apps we have to see the one that will aid us in achieving our aim.

Firstly, we will need to develop a frequency table to identify the most common genre for the two markets. This means we need frequency table for the `prime_genre` column for App Store dataset, and `Genres` and `Category` column for Google Play Store dataset.

### Part 2

We will need to create two functions to better represent the frequency of each genre for the two datasets:
    1. A function that generates the frequency table and shows the value for each genre in percentage
    2. Another function that exhibits the percentages in descending order.

In [17]:
def freq_table(dataset, index):     # First function to generate frequency tables
    column_dict = {}
    total = 0
    
    for each_element in dataset:
        if each_element[index] in column_dict:
            column_dict[each_element[index]] += 1
        else:
            column_dict[each_element[index]] = 1
            
    for each_element in column_dict:
        total += column_dict[each_element]
    
    for each_element in column_dict:
        column_dict[each_element] = column_dict[each_element] / total * 100
        
    return column_dict

In [18]:
def display_table(dataset, index):     # Second functin to sort the freq table in descending order
    table = freq_table(dataset, index)
    table_tuples = []
    
    for key in table:
        key_tuple = (table[key], key)
        table_tuples.append(key_tuple)
        
    sorted_table = sorted(table_tuples, reverse = True)
    
    for each_entry in sorted_table:
        print(each_entry[1], ':', each_entry[0])

Let's start by examining the prime_genre column for the App Store.

In [19]:
display_table(apple_final, 12) # Freq table for prime_genre(apple)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that about 58% of the English free apps are games. The next which is Entertainment is almost 8%, followed by Photo & Video which is approximately 5%. Only Educational apps comprise of 3.66%; Social networking amounts 3.29% of the dataset. Apps that belong to Reference, Business, Book, Navigation, Medical and Catalogs genre are few on App Store.

From this frequency table, it can be deduced that most English free apps on the App Store are built for entertainment purposes(games, photo and video, social networking, sports, music).

Though the number of apps for the game genre is significant, but that doesn't guarantee a large number of users for game apps, the supply might be more than demand, so the genre alone is not sufficient for our analysis.

Let's check the Category Column of Google Play Store.

In [20]:
display_table(google_final, 1) # Freq table for Category(google)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The Category shows that majority of the English free apps are categorized under family which is about 19%; next to the family category is game, which is approximately 9.7%, tools 8.5%, Business 4.6%. For the google play store category, it's correct to say that the difference between other categories apart from aforementioned is insignificant as compared to that of app store.

In [21]:
display_table(google_final, 9) # Freq table for Genres(google)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

For the genres result, most of the english free apps are under the tools genres which comprises of 8.4% of the whole dataset. This is preceded by Enterntainment genre, which is 6%,  then Education 5.3% and Business, approximately 5%. It is apparent that we have more genres (that is the categories are granular) in the Google Play Store dataset as compared to App Store dataset, this accounts for trivial differences between genres percentages in the gooogle play store.

Based on the result obtained from the `Category` and `Genre` column, we can deduce that the apps on google play store are balanced for both entertainment and practical purposes(education, shopping, utilities, productivity, lifestyle).

The difference between the `Genre` and `Category` is not really clear, but the `Category` column seems to present the bigger picture, which is what we will be working with.

## Most Popular Apps by Genre on AppStore

To find out the most popular app by genre we need to calculate the average number of installs for each app genre. This information can be found in the Google Play Store dataset under the `Install` column. In the App Store dataset, we don't have this information, but we can use the number of user ratings as a proxy, which can be found in the `rating_count_tot` column.

### Part 1

To calculate the number of installs per app genre for the App Store, we need to:
    1. Isolate the apps of each genre
    2. Find the sum of the user ratings for each genre
    3. Divide the sum by the number apps in that genre

Let's define a function to carry out the 3 aforementioned steps,

In [22]:
def installs_per_genre(freq_table, main_dataset, genre_index, installs_index): # function to show installs per app genre
    installs_table = []
    for genre in freq_table:
        total = 0
        len_genre = 0
        
        for app in main_dataset:
            genre_app = app[genre_index]
            if genre_app == genre:
                app_usr_ratings = float(app[installs_index])
                total += app_usr_ratings
                len_genre += 1
        
        avg_no_users = total / len_genre
        
        installs_per_genre_tuple = (avg_no_users, genre)
        installs_table.append(installs_per_genre_tuple)
        
    sorted_table = sorted(installs_table, reverse = True)
    
    for each_tuple in sorted_table:
        print(each_tuple[1], ':', each_tuple[0])

Let's generate the frequency table of App Store and show the number of installs per app gnere using the `installs_per_genre` function,

In [23]:
apple_app_genre = freq_table(apple_final, 12)
installs_per_genre(apple_app_genre, apple_final, 12, 6)

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


From the result obtained, let's explore each genre installs and see which one best suit or goal.

For effective analysis, we need to first sort the App store dataset with respect to number of `rating_count_tot` so we can see the number of installs for each app in a genre in descending order.

In [24]:
sorted_apple_final = []
installs_for_apple = []
for app in apple_final:
    app_tuple = (int(app[6]), app[0])
    installs_for_apple.append(app_tuple)
srtd_n_installs = sorted(installs_for_apple, reverse = True)
for each_tuple in srtd_n_installs:
    for each_app in apple_final:
        if(each_app[0] == each_tuple[1]) and (int(each_app[6]) == each_tuple[0]):
            sorted_apple_final.append(each_app)

In [25]:
for app in sorted_apple_final:
    if app[12] == 'Navigation':
        print(app[2], ':', app[6])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


For the `Navigation` genre, the number of installs are skewed towards the `Waze` and `Google Maps`, a newly developed app under this genre might not survive the competiton as we have for the `Railway Route Search`.

Next, let's check the `Reference` genre,

In [26]:
for app in sorted_apple_final:
    if app[12] == 'Reference':
        print(app[2], ':', app[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


For the `Reference` genre, the number of installs are a bit well distributed compared to `Navigation`, though some apps still have 0 installs. Let's check another genre,

In [30]:
for app in sorted_apple_final:
    if app[12] == 'Social Networking':
        print(app[2], ':', app[6])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

The `Social Networking` genre semms to be normally distributed, but the number of apps under this genre is much; the `Facebook` app has the majority of the installs. Let's explore some other genre and make a deduction from them:

In [31]:
for app in sorted_apple_final:
    if app[12] == 'Music':
        print(app[2], ':', app[6])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

In [32]:
for app in sorted_apple_final:
    if app[12] == 'Book':
        print(app[2], ':', app[6])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
謎解き : 0
謎解き2016 : 0
MangaTiara - love comic reader : 0
ikouhoushi : 0


In [33]:
for app in sorted_apple_final:
    if app[12] == 'Photo & Video':
        print(app[2], ':', app[6])

Instagram : 2161558
Snapchat : 323905
YouTube - Watch Videos, Music, and Live Streams : 278166
Pic Collage - Picture Editor & Photo Collage Maker : 123433
Funimate video editor: add cool effects to videos : 123268
musical.ly - your video social network : 105429
Photo Collage Maker & Photo Editor - Live Collage : 93781
Vine Camera : 90355
Google Photos - unlimited photo and video storage : 88742
Flipagram : 79905
Mixgram - Picture Collage Maker - Pic Photo Editor : 54282
Shutterfly: Prints, Photo Books, Cards Made Easy : 51427
Pic Jointer – Photo Collage, Camera Effects Editor : 51330
Color Pop Effects - Photo Editor & Picture Editing : 45320
Photo Grid - photo collage maker & photo editor : 40531
iSwap Faces LITE : 39722
MOLDIV - Photo Editor, Collage & Beauty Camera : 39501
Photo Editor by Aviary : 39501
Photo Lab: Picture Editor, effects & fun face app : 34585
Rookie Cam - Photo Editor & Filter Camera : 33921
FotoRus -Camera & Photo Editor & Pic Collage Maker : 32558
PicsArt Photo St

Among the genre we've explored, the `Music` genre has more apps but with well distributed number of installs, this means any newly developed app under this genre is given the potential to prosper provided it has attractive features.

Since the main goal is for users of our app to spend more time using our app, we can take the feature of a popular musical app and allow users to access and read the lyrics of the music they are listening to without leaving the music app and searching for the lyrics on a browser.

The `Weather` genre doesn't seem that interesting because people don't really spend time on whether app. The `Finance` and `Food & Drink` genre requires additional features that are outside the scope of this company such as, food delivery, finance expertise etc. Let's explore the google play store.

## Most Popular App By Genre on the Google Play Store


For the google dataset, we have data about the number of installs, with this, we should be able to get aclearer picture of the most popular app by genre. However, the values under the `install` column are open-ended(100+, 1,000+, 5,000+ etc). For example, we don't know if an with 100+ installs have 101, 200 or 500 installs, but we don't really need precise data for our analysis, we only want to know which app genres attract most installs, so we will leave values as they are taht is, we will assume that an app with 100+ installs have 100 installs.

To perform our computation however, we need to convert the values of the installs to float; to do this, we need to remove the commas `,` and the plus `+` character from the install values using the `str.replace(old, new)` function.

In [40]:
google_app_genre = freq_table(google_final, 1)

install_freq_table = []

for category in google_app_genre:
    total = 0
    len_category = 0
    
    for each_app in google_final:
        category_app = each_app[1]
        if category_app == category:
            cat_string = each_app[5].replace('+', '')
            cat_string = cat_string.replace(',', '')
            
            cat_float = float(cat_string)
            
            total += cat_float
            len_category +=1
    
    avg_no_user = total / len_category
    
    cat_tuple = (avg_no_user, category)
    install_freq_table.append(cat_tuple)
    
srted_install_table = sorted(install_freq_table, reverse = True)

for each_cat in srted_install_table:
    print(each_cat[1], each_cat[0])

COMMUNICATION 38456119.167247385
VIDEO_PLAYERS 24727872.452830188
SOCIAL 23253652.127118643
PHOTOGRAPHY 17840110.40229885
PRODUCTIVITY 16787331.344927534
GAME 15588015.603248259
TRAVEL_AND_LOCAL 13984077.710144928
ENTERTAINMENT 11640705.88235294
TOOLS 10801391.298666667
NEWS_AND_MAGAZINES 9549178.467741935
BOOKS_AND_REFERENCE 8767811.894736841
SHOPPING 7036877.311557789
PERSONALIZATION 5201482.6122448975
WEATHER 5074486.197183099
HEALTH_AND_FITNESS 4188821.9853479853
MAPS_AND_NAVIGATION 4056941.7741935486
FAMILY 3695641.8198090694
SPORTS 3638640.1428571427
ART_AND_DESIGN 1986335.0877192982
FOOD_AND_DRINK 1924897.7363636363
EDUCATION 1833495.145631068
BUSINESS 1712290.1474201474
LIFESTYLE 1437816.2687861272
FINANCE 1387692.475609756
HOUSE_AND_HOME 1331540.5616438356
DATING 854028.8303030303
COMICS 817657.2727272727
AUTO_AND_VEHICLES 647317.8170731707
LIBRARIES_AND_DEMO 638503.734939759
PARENTING 542603.6206896552
BEAUTY 513151.88679245283
EVENTS 253542.22222222222
MEDICAL 120550.6198083

We can see that the `COMMUNICATION` category has the most number of installs; but we can't just assume it to be the best category for our app, we need to check the distribution of users for each categories.
First, let's sort the Google Play Store dataset with respect to `installs` for easy analysis.

In [43]:
sorted_google_final = []
installs_for_google = []

for app in google_final:
    cat_str = app[5].replace('+', '')
    cat_str = cat_str.replace(',', '')
    cat_flt = float(cat_str)
    app_tuple = (cat_flt, app[0])
    installs_for_google.append(app_tuple)
srtd_google_installs = sorted(installs_for_google, reverse = True)

for each_tuple in srtd_google_installs:
    
    for each_app in google_final:
        
        cat_str_com = each_app[5].replace('+', '')
        cat_str_com = cat_str_com.replace(',', '')
        cat_flt_com = float(cat_str_com)
        
        if(each_app[0] == each_tuple[1]) and (cat_flt_com == each_tuple[0]):
            sorted_google_final.append(each_app)

In [45]:
for app in sorted_google_final:
    if app[1] == 'COMMUNICATION':
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
Skype - free IM & video calls : 1,000,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
Hangouts : 1,000,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Gmail : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Viber Messenger : 500,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
imo beta free calls and text : 100,000,000+
Yahoo Mail – Stay Organized : 100,000,000+
Who : 100,000,000+
WeChat : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
Telegram : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,0

Number of installs in the `COMMUNICATION` category are skewed up by apps with installs of `1,000,000,000+`, `500,000,000+`, and `100,000,000+`. This means the competition in this category is very tough; it's unlikely that a newly developed app will prosper at this category.

Let's explore other categories and see the distribution of the installs:

In [46]:
for app in sorted_google_final:
    if app[1] == 'VIDEO_PLAYERS':
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
VLC for Android : 100,000,000+
Motorola Gallery : 100,000,000+
Motorola FM Radio : 100,000,000+
Dubsmash : 100,000,000+
Vote for : 50,000,000+
Vigo Video : 50,000,000+
VMate : 50,000,000+
Samsung Video Library : 50,000,000+
Ringdroid : 50,000,000+
MiniMovie - Free Video and Slideshow Editor : 50,000,000+
LIKE – Magic Video Maker & Community : 50,000,000+
KineMaster – Pro Video Editor : 50,000,000+
HD Video Downloader : 2018 Best video mate : 50,000,000+
DU Recorder – Screen Recorder, Video Editor, Live : 50,000,000+
video player for android : 10,000,000+
iMediaShare – Photos & Music : 10,000,000+
YouTube Studio : 10,000,000+
Video Player All Format : 10,000,000+
Video Downloader - for Instagram Repost App : 10,000,000+
Video Downloader : 10,000,000+
Ustream : 10,000,000+
Quik – 

In [49]:
for app in sorted_google_final:
    if app[1] == 'PRODUCTIVITY':
        print(app[0], ':', app[5])

Google Drive : 1,000,000,000+
Microsoft Word : 500,000,000+
Google Calendar : 500,000,000+
Dropbox : 500,000,000+
Cloud Print : 500,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Samsung Notes : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneNote : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft Excel : 100,000,000+
Google Slides : 100,000,000+
Google Sheets : 100,000,000+
Google Keep : 100,000,000+
Google Docs : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
ES File Explorer File Manager : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
CamScanner - Phone PDF Creator : 100,000,000+
Adobe Acrobat Reader : 100,000,000+
myAT&T : 50,000,000+
Verizon Cloud : 50,000,000+
QR Droid : 50,000,000+
My Airtel-Online Recharge, Pay Bill, Wallet, UPI : 50,000,000+
Mobizen Screen Recorder - Record, Capture, Edit : 50,000,000+
MEGA : 50,000,000+
File Bro

In [50]:
for app in sorted_google_final:
    if app[1] == 'ENTERTAINMENT':
        print(app[0], ':', app[5])

Talking Ben the Dog : 100,000,000+
Talking Angela : 100,000,000+
Netflix : 100,000,000+
IMDb Movies & TV : 100,000,000+
Hotstar : 100,000,000+
Twitch: Livestream Multiplayer Games & Esports : 50,000,000+
Talking Ginger 2 : 50,000,000+
PlayStation App : 50,000,000+
Amazon Prime Video : 50,000,000+
ivi - movies and TV shows in HD : 10,000,000+
WWE : 10,000,000+
Vudu Movies & TV : 10,000,000+
Viki: Asian TV Dramas & Movies : 10,000,000+
Tubi TV - Free Movies & TV : 10,000,000+
SketchBook - draw and paint : 10,000,000+
STARZ : 10,000,000+
Redbox : 10,000,000+
Movies by Flixster, with Rotten Tomatoes : 10,000,000+
Motorola Spotlight Player™ : 10,000,000+
Mobile TV : 10,000,000+
MEGOGO - Cinema and TV : 10,000,000+
Imgur: Find funny GIFs, memes & watch viral videos : 10,000,000+
Fandango Movies - Times + Tickets : 10,000,000+
FOX : 10,000,000+
Crunchyroll - Everything Anime : 10,000,000+
Crackle - Free TV & Movies : 10,000,000+
Colorfy: Coloring Book for Adults - Free : 10,000,000+
CBS - Ful

All the categories we've explored so far seem to have some popular apps that skew up the number of installs per genre except for the `ENTERTAINMENT` genre which has it's topmost app `Talking Ben the Dog` to have +100 million installs.

And this `ENTERTAINMENT` catregory mostly comprises of TV apps, with very few number of musical apps.

Therefore, a musical app with lyrics feature has an outstanding chance of success for both markets.

## In conclusion

We analysed the App Store and Google Play Store dataset with the aim of providing our developers with the best app profile that can be profitable for both markets.

We concluded that a musical app with lyrics features will be most profitable for both markets since the musical apps in App Store have the potential to prosper as long as they have attractive features, and users will tend to spend more time on the apps reading the lyrics, which is where we genrate revenue from, while on the Google Play Store, we have very few musical apps with high number of installs, this will make a good condition for a newly developed musical apps with lyrics.